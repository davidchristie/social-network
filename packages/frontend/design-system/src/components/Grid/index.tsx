import MaterialGrid, { GridProps } from "@material-ui/core/Grid";
import React from "react";

const Grid: React.StatelessComponent<GridProps> = props => {
  return (
    <MaterialGrid {...props} />
  );
};

export default Grid;
