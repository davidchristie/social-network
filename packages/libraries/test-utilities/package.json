{
  "name": "test-utilities",
  "private": true,
  "scripts": {
    "build": "rimraf dist && tsc"
  },
  "version": "1.0.0"
}
