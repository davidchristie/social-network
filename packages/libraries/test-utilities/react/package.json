{
  "main": "../dist/react/index.js",
  "name": "test-utilities/react",
  "private": true,
  "version": "1.0.0"
}
