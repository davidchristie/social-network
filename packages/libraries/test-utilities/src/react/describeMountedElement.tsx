import { mount } from "enzyme";
import React from "react";

export default function describeMountedElement (
  name: string,
  element: React.ReactElement<any>
) {
  describe(name, () => {
    it("renders without errors", () => {
      mount(element);
    });

    it("matches shapshot", () => {
      const wrapper = mount(element);
      expect(wrapper).toMatchSnapshot();
    });
  });
}
