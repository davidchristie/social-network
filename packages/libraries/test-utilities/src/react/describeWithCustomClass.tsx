import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

interface Props {
  className?: string;
}

export default function describeWithCustomClass (
  Component: React.ComponentType<Props>
) {
  describe("with custom class", () => {
    const CUSTOM_CLASS = "CUSTOM_CLASS";

    let wrapper: ShallowWrapper;

    beforeEach(() => {
      wrapper = shallow(
        <Component className={CUSTOM_CLASS} />,
      );
    });

    it("has custom class", () => {
      expect(wrapper.hasClass(CUSTOM_CLASS)).toBe(true);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
}
