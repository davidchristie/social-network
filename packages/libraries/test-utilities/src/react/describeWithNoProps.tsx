import React from "react";

import describeWithProps from "./describeWithProps";

export default function describeWithNoProps (
  Component: React.ComponentType
) {
  describeWithProps("with no props", Component, {});
}
