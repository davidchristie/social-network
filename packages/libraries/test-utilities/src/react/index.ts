export { default as describeMountedElement } from "./describeMountedElement";
export { default as describeWithCustomClass } from "./describeWithCustomClass";
export { default as describeWithNoProps } from "./describeWithNoProps";
export { default as describeWithProps } from "./describeWithProps";
export { default as itMatchesSnapshot } from "./itMatchesSnapshot";
