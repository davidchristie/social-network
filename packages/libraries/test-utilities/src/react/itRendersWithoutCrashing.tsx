import { mount } from "enzyme";
import React from "react";
import { MockedProvider } from "react-apollo/test-utils";

export default function itRendersWithoutCrashing (Component: React.ComponentType) {
  it("renders without crashing", () => {
    mount(
      <MockedProvider>
        <Component />
      </MockedProvider>
    );
  });
}
