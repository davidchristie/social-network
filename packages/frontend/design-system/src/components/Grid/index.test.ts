import { describeWithNoProps } from "test-utilities/react";
import Grid from ".";

describe("Grid component", () => {
  describeWithNoProps(Grid);
});
