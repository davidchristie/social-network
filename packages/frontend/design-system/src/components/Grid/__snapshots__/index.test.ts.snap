// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component with no props matches snapshot 1`] = `<WithStyles(Grid) />`;
